### PR TITLE
Implement GASVISCT for black-oil [Bug fix]

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**
GASVISCT is a E300 keyword and requires COMPS keyword. OPM current implementation would simply give this error:
image
That would happen because the implementation was trying to access the content of COMPS that was not there and there was no safeguard for that. 
I went ahead and implemented this keyword for a black-oil gas, considering we don't have support compositional formulation.  In the future it can be extended for compositional. 
Note, for now I just assumed references RV and RVW = 0, we could add these two parameters to the keyword VISCREF or create a new keyword for gas. What do you think @OPMUSER?
PS: I will add a test case in opm-tests today.

### CODE REVIEW

File: src/opm/input/eclipse/EclipseState/Tables/Tables.cpp [hunk 1]
```diff
@@ -1170,46 +1170,12 @@ WatvisctTable::getWaterViscosityColumn() const
     return SimpleTable::getColumn(1);
 }
 
-GasvisctTable::GasvisctTable(const Deck& deck, const DeckItem& deckItem)
+GasvisctTable::GasvisctTable(const DeckItem& item, const int tableID)
 {
-    int numComponents = deck.get<ParserKeywords::COMPS>().back().getRecord(0).getItem(0).get<int>(0);
```
Comment (goncalvesmachadoc) [inline src/opm/input/eclipse/EclipseState/Tables/Tables.cpp:1175]: Here it was the main issue.

File: src/opm/material/fluidsystems/blackoilpvt/GasPvtThermal.cpp [hunk 2]
```diff
@@ -61,14 +61,41 @@ initFromState(const EclipseState& eclState, const Schedule& schedule)
...
-            gasvisctCurves_[regionIdx].setXYContainers(T, mu);
+            const auto& TCol = gasvisctTables[regionIdx].getColumn("Temperature").vectorCopy();
+            const auto& muCol = gasvisctTables[regionIdx].getColumn("Viscosity").vectorCopy();
+            gasvisctCurves_[regionIdx].setXYContainers(TCol, muCol);
+
+            viscrefPress_[regionIdx] = viscrefTable[regionIdx].reference_pressure;
+
+            // temperature used to calculate the reference viscosity [K]. 
+            constexpr const Scalar Tref = 273.15 + 20;
+            constexpr const Scalar Rvref = 0.0;
```
Comment (totto82) [inline src/opm/material/fluidsystems/blackoilpvt/GasPvtThermal.cpp:90]: Maybe add a comment that Tref is not used? I was a bit puzzled by why 20C was used here.

**Review by totto82 — APPROVED** [review]
Thanks for the fix.

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

**Pull request merged**